plugins/utils/ollama.nix

```diff
@@ -0,0 +1,226 @@
+        The default model to use.
+      '';
+
+      prompts = let
+        promptOptions = {
+          prompt = mkOption {
+            type = types.str;
+            description = ''
+              The prompt to send to the model.
+
+              Replaces the following tokens:
+              - `$input`: The input from the user
+              - `$sel`: The currently selected text
+              - `$ftype`: The filetype of the current buffer
+              - `$fname`: The filename of the current buffer
+              - `$buf`: The contents of the current buffer
+              - `$line`: The current line in the buffer
+              - `$lnum`: The current line number in the buffer
+            '';
+          };
+
+          inputLabel = helpers.defaultNullOpts.mkStr "> " ''
+            The label to use for an input field.
+          '';
+
+          action = helpers.mkNullOrOption actionOptionType ''
+            How to handle the output.
+
+            See [here](https://github.com/nomnivore/ollama.nvim/tree/main#actions) for more details.
+
+            Defaults to the value of `plugins.ollama.action`.
+          '';
+
+          model = helpers.mkNullOrOption types.str ''
+            The model to use for this prompt.
+
+            Defaults to the value of `plugins.ollama.model`.
+          '';
+
+          extract =
+            helpers.defaultNullOpts.mkNullable
+            (with types; either str (enum [false]))
+            "```$ftype\n(.-)```"
+            ''
+              A `string.match` pattern to use for an Action to extract the output from the response
+              (Insert/Replace).
+            '';
+
+          options = helpers.mkNullOrOption (with types; attrsOf anything) ''
+            Additional model parameters, such as temperature, listed in the documentation for the [Modelfile](https://github.com/jmorganca/ollama/blob/main/docs/modelfile.md#valid-parameters-and-values).
+          '';
+
+          system = helpers.mkNullOrOption (helpers.nixvimTypes.maybeRaw types.str) ''
+            The SYSTEM instruction specifies the system prompt to be used in the Modelfile template,
+            if applicable.
+            (overrides what's in the Modelfile).
+          '';
+
+          format = helpers.defaultNullOpts.mkEnumFirstDefault ["json"] ''
+            The format to return a response in.
+            Currently the only accepted value is `"json"`.
+          '';
+        };
+
+        processPrompt = prompt:
+          if isAttrs prompt
+          then {
+            inherit (prompt) prompt;
+            input_label = prompt.inputLabel;
+            inherit
+              (prompt)
+              action
+              model
+              extract
+              options
+              system
+              format
+              ;
+          }
+          else prompt;
+      in
+        mkOption {
+          type = with types;
+            attrsOf (
+              either
+              (submodule {
```

Comment:
I think doing the either means the documentation won't render the prompt options :/

Reply:
Maybe, but we don't really have the choice here...